### PR TITLE
unused arg within user defined semiring, cleaner build output, faster …

### PR DESCRIPTION
**Pull request by swilly22**
…build time

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -99,10 +109,13 @@ $(LIBTRIEMAP):
 .PHONY: $(LIBTRIEMAP)
 
 BLAS_CFLAGS = -fPIC -I $(abs_path)/util -include rmalloc.h -DREDIS_MODULE_TARGET -DGB_MALLOC=rm_malloc -DGB_CALLOC=rm_calloc -DGB_REALLOC=rm_realloc -DGB_FREE=rm_free -DGBCOMPACT
-$(GRAPHBLAS):
-	$(MAKE) CMAKE_OPTIONS="-DJOBS=32 -DUSER_POSIX=1 -DCMAKE_CXX_FLAGS=-std=c++11 -DCMAKE_C_FLAGS='$(BLAS_CFLAGS)'" static_only -C ../deps/GraphBLAS
 
-.PHONY: $(GRAPHBLAS)
```
Comment (jeffreylovitz) [inline src/Makefile]: Nice catch, this line was wasting a lot of time.

File: src/Makefile [hunk 2]
```diff
@@ -99,10 +109,13 @@ $(LIBTRIEMAP):
...
+# The target libgraphblas.a required by redisgraph.so
+# is conditionaly dependent on the existence of the library libgraphblas.a
+# incase the file doesn't exists we'll build it, otherwise this rule is skipped.
+# see Makefile "order-only dependencies".
+$(GRAPHBLAS): | $(GRAPHBLAS)
```
Comment (jeffreylovitz) [inline src/Makefile]: This doesn't work for me:
`make[1]: Circular ../deps/GraphBLAS/build/libgraphblas.a <- ../deps/GraphBLAS/build/libgraphblas.a dependency dropped.`

File: tests/unit/Makefile [hunk 3]
```diff
@@ -69,7 +79,7 @@ build: $(TEST_OBJECTS) $(TEST_EXECUTABLES) $(DEPS)
 
 test: build
 	for t in $(TEST_EXECUTABLES); do \
-		./$$t || exit 1; \
+		./$$t; \
```
Comment (jeffreylovitz) [inline tests/unit/Makefile]: Accidental deletion?

Comment (swilly22) [reply tests/unit/Makefile]: yep, sorry about that

File: src/Makefile (hunk 2 above)
Comment (swilly22) [reply src/Makefile]: same here, let's discus

**Pull request merged**